cmd/oras/root/cp.go

```diff
@@ -191,3 +195,40 @@ func runCopy(ctx context.Context, opts copyOptions) error {
 
 	return nil
 }
+
+// RecursiveCopy copies an artifact and its referrers from one target to another.
+// If the artifact is a manifest list or index, referrers of its manifests are copied as well.
+func RecursiveCopy(ctx context.Context, src oras.ReadOnlyGraphTarget, dst oras.Target, dstRef string, root ocispec.Descriptor, opts oras.ExtendedCopyOptions) error {
+	var err error
+	if root.MediaType == ocispec.MediaTypeImageIndex || root.MediaType == docker.MediaTypeManifestList {
+		var fetched []byte
+		fetched, err := content.FetchAll(ctx, src, root)
+		if err != nil {
+			return err
+		}
+		var index ocispec.Index
+		if err = json.Unmarshal(fetched, &index); err != nil {
+			return nil
+		}
+
+		var wg sync.WaitGroup
+		for _, desc := range index.Manifests {
+			wg.Add(1)
+			go func(desc ocispec.Descriptor) {
+				defer wg.Done()
+				err = oras.ExtendedCopyGraph(ctx, src, dst, desc, opts.ExtendedCopyGraphOptions)
+			}(desc)
+		}
+		wg.Wait()
+		if err != nil {
```

Comment:
Can this line catch the `err` in line 219?

Reply:
go routine doesn't support returning error directly.

Reply:
It looks like the error returned may be overwritten, if we write the error handling this way.

Reply:
indeed, I have changed implementation, PTAL